pymongo/pool.py

```diff
@@ -1266,13 +1269,13 @@ def connect(self, all_credentials=None):
             raise
 
         sock_info = SocketInfo(sock, self, self.address, conn_id)
-        if self.handshake:
-            sock_info.ismaster(all_credentials)
-            self.is_writable = sock_info.is_writable
-
         try:
+            if self.handshake:
+                sock_info.ismaster(all_credentials)
+                self.is_writable = sock_info.is_writable
+
             sock_info.check_auth(all_credentials)
-        except Exception:
+        except:  # Catch all exceptions.
```

Comment:
What was not being caught without the bare `Except`?

Reply:
Things that inherit from BaseException, like gevent.Timeout. I'll undo this change and add it in when merging the gevent PR.

Reply:
Thanks.